VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room.component.ts

```diff
@@ -0,0 +1,705 @@
+            participant: this.vhParticipant.id,
+            roomLabel: roomLabel
+        });
+        this.hasTriedToLeaveConsultation = false;
+        await this.consultationService.joinPrivateConsultationRoom(this.vhConference.id, this.vhParticipant.id, roomLabel);
+        this.closeJoinPrivateConsultationModal();
+        this.privateConsultationAccordianExpanded = false;
+    }
+
+    async setRoomLock(lock: boolean) {
+        const roomLabel = this.vhParticipant.room?.label;
+        if (!roomLabel) {
+            return;
+        }
+
+        this.logger.info(`${this.componentLoggerPrefix} attempting to set room lock state`, {
+            conference: this.vhConference?.id,
+            participant: this.vhParticipant.id,
+            roomLabel: roomLabel,
+            lock: lock
+        });
+        await this.consultationService.lockConsultation(this.vhConference.id, roomLabel, lock);
+    }
+
+    closeStartPrivateConsultationModal() {
+        this.displayStartPrivateConsultationModal = false;
+    }
+
+    closeJoinPrivateConsultationModal() {
+        this.displayJoinPrivateConsultationModal = false;
+    }
+
+    toggleAccordian() {
+        this.privateConsultationAccordianExpanded = !this.privateConsultationAccordianExpanded;
+    }
+
+    setTrapFocus() {
+        ModalTrapFocus.trap('video-container');
+    }
+
+    dismissJoinHearingWarning() {
+        this.showJoinHearingWarning = false;
+        this.setUpSubscribers();
+    }
+
+    onLeaveHearingButtonClicked() {
+        this.displayLeaveHearingPopup = true;
+    }
+
+    leave(confirmation: boolean) {
+        this.displayLeaveHearingPopup = false;
+        if (!confirmation) {
+            return;
+        }
+        const feedbackUrl = 'https://www.smartsurvey.co.uk/s/VideoHearings_Feedback/';
+        window.location.assign(feedbackUrl);
+    }
+
+    init() {
+        this.divTrapId = 'video-container';
+
+        if (this.userRole === WaitingRoomUserRole.Judge) {
+            this.initForJudge();
+            return;
+        }
+
+        this.destroyedSubject = new Subject();
+
+        this.unloadDetectorService.shouldUnload.pipe(takeUntil(this.destroyedSubject)).subscribe(() => this.onShouldUnload());
+        this.unloadDetectorService.shouldReload.pipe(take(1)).subscribe(() => this.onShouldReload());
+
+        this.errorCount = 0;
+        this.logger.debug(`${this.componentLoggerPrefix} loading waiting room`);
+        this.connected = false;
+        this.getConference();
+        if (this.deviceTypeService.isHandheldIOSDevice()) {
+            this.showJoinHearingWarning = true;
+        } else {
+            this.setUpSubscribers();
+        }
+    }
+
+    initForJudge() {
+        if (!this.vhConference || !this.vhParticipant) {
+            return;
+        }
+
+        this.errorCount = 0;
+        this.logger.debug(`${this.componentLoggerPrefix} Loading judge waiting room`);
+        this.loggedInUser = this.route.snapshot.data['loggedUser'];
```

Comment:
invoke `this.conferenceStore.dispatch(AuthActions.loadUserProfile());` in `ngOnInit`. There's already an effect that populates the profile into the redux store so you can simply use the `getUserProfile` selector. See `ngOnInit` in `BaseSelfTestComponentDirective`

Reply:
turns out `loggedInUser` isn't actually referenced anywhere, so have removed it